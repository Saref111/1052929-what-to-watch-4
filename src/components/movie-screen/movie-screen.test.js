import React from 'react';
import renderer from "react-test-renderer";
import MovieScreen from "./movie-screen.jsx";

describe(`MovieScreen snaptests`, () => {
  test(`Snapshot MovieScreen test`, () => {
    const tree = renderer.create(
        <MovieScreen
          getProgress={() => {}}
          toggleFullScreen={() => {}}
          handlePlayPause={() => {}}
          toggleMovieScreenHandler={() => {}}
          currentPosition={44}
          isFullscreen={false}
          isPlaying={true}
          movieLength={100}
          movieSrc={`111111`}
          poster={`12345`}
        />, {
          createNodeMock: () => {
            return {
              poster: null,
              src: null,
              muted: null,
            };
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test(`Snapshot MovieScreen test fullscreen mode`, () => {
    const tree = renderer.create(
        <MovieScreen
          getProgress={() => {}}
          toggleFullScreen={() => {}}
          handlePlayPause={() => {}}
          toggleMovieScreenHandler={() => {}}
          currentPosition={44}
          isFullscreen={true}
          isPlaying={true}
          movieLength={100}
          movieSrc={`111111`}
          poster={`12345`}
        />, {
          createNodeMock: () => {
            return {
              oncanplaythrough: null,
              onpause: null,
              onplay: null,
              ontimeupdate: null,
            };
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


