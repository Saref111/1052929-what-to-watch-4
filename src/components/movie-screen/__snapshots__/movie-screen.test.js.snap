// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieScreen snaptests Snapshot MovieScreen test 1`] = `
<div
  className="fullscreen"
>
  <div
    className="player"
  >
    <video
      className="player__video"
      crossOrigin="anonymous"
    />
    <a
      className="player__exit"
      href="/films/undefined"
      onClick={[Function]}
      type="button"
    >
      Exit
    </a>
    <div
      className="player__controls"
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max="100"
            value={44}
          />
          <div
            className="player__toggler"
            style={
              Object {
                "left": "44%",
              }
            }
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          0 : 44
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="21"
            viewBox="0 0 14 21"
            width="14"
          >
            <use
              xlinkHref="#pause"
            />
          </svg>
          <span>
            Pause
          </span>
        </button>
        <div
          className="player__name"
        >
          Transpotting
        </div>
        <button
          className="player__full-screen"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieScreen snaptests Snapshot MovieScreen test fullscreen mode 1`] = `
<div
  className="fullscreen"
>
  <div
    className="player"
  >
    <video
      className="player__video"
      crossOrigin="anonymous"
    />
    <a
      className="player__exit"
      href="/films/undefined"
      onClick={[Function]}
      type="button"
    >
      Exit
    </a>
    <div
      className="player__controls"
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max="100"
            value={44}
          />
          <div
            className="player__toggler"
            style={
              Object {
                "left": "44%",
              }
            }
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          0 : 44
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="21"
            viewBox="0 0 14 21"
            width="14"
          >
            <use
              xlinkHref="#pause"
            />
          </svg>
          <span>
            Pause
          </span>
        </button>
        <div
          className="player__name"
        >
          Transpotting
        </div>
        <button
          className="player__full-screen"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
